src/plugins/walletData.ts

```diff
@@ -24,4 +29,34 @@ export const walletData: iWalletWrapper = {
       internalWalletData.accountState = val.accountState;
     }
   },
+
+  clear: () => {
+    internalWalletData.syncWallet = undefined;
+    internalWalletData.accountState = undefined;
+  },
+
+  syncProvider: {
+    async load() {
+      if (internalWalletData.syncProvider) {
+        return;
+      }
+      providerPromise = getDefaultProvider(ZK_NETWORK as Network, "HTTP");
+      try {
+        await new Promise((resolve) => {
+          setTimeout(() => {
+            resolve(undefined);
+          }, 3000);
+        });
+        internalWalletData.syncProvider = await providerPromise;
+      } catch (error) {
+        console.log("Failed to load provider", error);
+      }
+    },
+    async get() {
+      if (!internalWalletData.syncProvider || typeof providerPromise === "undefined") {
+        walletData.syncProvider.load();
```

Comment:
because there is await right under it.
The process is:
1. Create a promise for provider
2. Return promise
3. Anything that requests the promise can await for it, if it was already loaded then its just instantly continues.
So here we check if we requested the promise and if we didn't then we request it and return this promise